### DISCUSSION TITLE
Can I use File extension to download base64 files？

### DISCUSSION THREAD

**Discussion by MA-master-sec** [TurboWarp/extensions#776] [General]
Although it can upload,but it can't download base64..

**Comment by CST1229** [comment]
It can, with the download URL block:
![image](https://github.com/TurboWarp/extensions/assets/68464103/bca46e24-b893-4c8e-8c86-df9c3c9aa58a)
(you can just prepend something like `data:application/octet-stream;base64,` to the start if you only have base64 data and not an actual data URL)

**Reply by yuri-kiss** [reply]
appending `base64,` to the output will attempt to convert it before its downloaded :|

**Reply by yuri-kiss** [reply]
if you want it converted than you would add it lol

**Reply by yuri-kiss** [reply]
if you dont wanna convert it just remove the `base64,` part

**Reply by CST1229** [reply]
then you can just use the plain `download` block

**Reply by yuri-kiss** [reply]
exactly

**Comment by MA-master-sec** [comment]
![image](https://github.com/TurboWarp/extensions/assets/126303227/cf86ec1b-1181-46a9-bedc-ba8286d9c9d9)
where?

**Reply by CST1229** [reply]
Are you using the desktop app? I think that has an outdated version of the extension without that block (you'll have to wait).

**Reply by MA-master-sec** [reply]
O.
yes the browser extension have it
